artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ConnectionView.java

```diff
@@ -91,7 +95,14 @@ public Object getField(RemotingConnection connection, String fieldName) {
          case "protocol":
             return connection.getProtocolName();
          case "clientID":
-            return connection.getClientID();
+            String clientID = null;
+            for (ServerSession session : sessions) {
+               if (clientID == null) {
+                  clientID = session.getMetaData("jms-client-id") == null ? connection.getClientID() : session.getMetaData("jms-client-id");
```

Comment:
jms specifics should not be in core broker. Also this is the connection view, this data is off the session view, so should be visible there, not on the connection.